File: frontend/src/objects/merchant/store/StoreMetrics.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AverageRating,
  CurrencyCents,
  EntityCount,
} from '@/objects/core/SharedObjects'

export type StoreMetrics = {
  averageRating: AverageRating
  ratingCount: EntityCount
  oneStarRatingCount: EntityCount
  revenueCents: CurrencyCents
}

File: backend/src/main/scala/services/merchant/objects/StoreMetrics.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class StoreMetrics(
    averageRating: AverageRating,
    ratingCount: EntityCount,
    oneStarRatingCount: EntityCount,
    revenueCents: CurrencyCents,
)
object StoreMetrics:
  given Encoder[StoreMetrics] = deriveEncoder
  given Decoder[StoreMetrics] = deriveDecoder
